src/lib/sys/configuration.h

```diff
@@ -42,41 +33,31 @@ typedef struct ConfigPartition {
     ConfigValue_t values[MAX_NUM_KV];
 }__attribute__((packed, aligned(1))) ConfigPartition_t;
 
-class Configuration : public AbstractConfiguration {
-public:
-    Configuration(NvmPartition &flash_partition, uint8_t *ram_partition, size_t ram_partition_size);
-    bool getConfig(const char * key, size_t key_len, uint32_t &value);
-    bool getConfig(const char * key, size_t key_len, int32_t &value);
-    bool getConfig(const char * key, size_t key_len, float &value);
-    bool getConfig(const char * key, size_t key_len, char *value, size_t &value_len);
-    bool getConfig(const char * key, size_t key_len, uint8_t *value, size_t &value_len);
-    bool getConfigCbor(const char * key, size_t key_len, uint8_t *value, size_t &value_len);
-    bool setConfig(const char * key, size_t key_len, uint32_t value);
-    bool setConfig(const char * key, size_t key_len, int32_t value);
-    bool setConfig(const char * key, size_t key_len, float value);
-    bool setConfig(const char * key, size_t key_len, const char *value, size_t value_len);
-    bool setConfig(const char * key, size_t key_len, const uint8_t *value, size_t value_len);
-    const ConfigKey_t* getStoredKeys(uint8_t &num_stored_keys);
-    bool setConfigCbor(const char * key, size_t key_len, uint8_t *value, size_t value_len);
-    bool removeKey(const char * key, size_t key_len);
-    static const char* dataTypeEnumToStr(ConfigDataTypes_e type);
-    bool saveConfig(bool restart=true);
-    bool getValueSize(const char * key, size_t key_len, size_t &size);
-    bool needsCommit(void);
-    static bool cborTypeToConfigType(const CborValue *value, ConfigDataTypes_e &configType);
-private:
-    bool findKeyIndex(const char * key, size_t len, uint8_t &idx);
-    bool prepareCborParser(const char * key, size_t key_len, CborValue &it, CborParser &parser);
-    bool prepareCborEncoder(const char * key, size_t key_len, CborEncoder &encoder, uint8_t &keyIdx, bool &keyExists);
-    bool loadAndVerifyNvmConfig(void);
-
-    static constexpr uint32_t CONFIG_START_OFFSET_IN_BYTES = 0;
-    static constexpr uint32_t CONFIG_LOAD_TIMEOUT_MS = 5000;
+typedef bool (*ConfigRead)(BmConfigPartition partition, uint32_t offset, uint8_t *buffer, size_t length, uint32_t timeout_ms);
+typedef bool (*ConfigWrite)(BmConfigPartition partition, uint32_t offset, uint8_t *buffer, size_t length, uint32_t timeout_ms);
+typedef void (*ConfigRestart)(void);
+
+void config_init(ConfigRead read, ConfigWrite write, ConfigRestart restart);
```

Comment:
The only change I'd recommend is passing a struct like your `ConfigCb` that holds the 3 callbacks rather than 3 separate params, because it would allow us to add and remove callbacks in the future without changing the signature of the function.

Reply:
Yea I agree! Good callout!

Reply:
This API now takes 0 arguments due to not using callbacks anymore.